aws/resource_aws_cognito_user_pool.go

```diff
@@ -926,22 +938,18 @@ func resourceAwsCognitoUserPoolUpdate(d *schema.ResourceData, meta interface{})
 		params.SmsVerificationMessage = aws.String(v.(string))
 	}
 
-	if v, ok := d.GetOk("tags"); ok {
-		params.UserPoolTags = tagsFromMapGeneric(v.(map[string]interface{}))
```

Comment:
Since the API provides a tags overwriting field during update, we should probably stick with it, instead of introducing our own extra API calls with `keyvaluetags.CognitoidentityproviderUpdateTags` 👍 

Let's remove the new code above:

```go
	arn := d.Get("arn").(string)
	if d.HasChange("tags") {
		o, n := d.GetChange("tags")

		if err := keyvaluetags.CognitoidentityproviderUpdateTags(conn, arn, o, n); err != nil {
			return fmt.Errorf("error updating Cognito Identity Pool (%s) tags: %s", arn, err)
		}
	}
```

And refactor this logic here to:

```go
	if v, ok := d.GetOk("tags"); ok {
		params.UserPoolTags = keyvaluetags.New(v.(map[string]interface{})).IgnoreAws().CognitoidentityproviderTags()
	}
```

Reply:
fixed, i'm not sure this is done uniformly in all resources though.

Reply:
--- PASS: TestAccAWSCognitoUserPool_basic (45.42s)
--- PASS: TestAccAWSCognitoUserPool_withAdminCreateUserConfiguration (95.15s)
--- PASS: TestAccAWSCognitoUserPool_withAdvancedSecurityMode (113.08s)
--- PASS: TestAccAWSCognitoUserPool_withDeviceConfiguration (88.51s)
--- PASS: TestAccAWSCognitoUserPool_withEmailVerificationMessage (76.93s)
--- PASS: TestAccAWSCognitoUserPool_withSmsVerificationMessage (83.78s)
--- SKIP: TestAccAWSCognitoUserPool_withEmailConfiguration (0.00s)
--- PASS: TestAccAWSCognitoUserPool_withSmsConfiguration (73.77s)
--- PASS: TestAccAWSCognitoUserPool_withSmsConfigurationUpdated (110.47s)
--- PASS: TestAccAWSCognitoUserPool_withTags (115.48s)
--- PASS: TestAccAWSCognitoUserPool_withAliasAttributes (81.12s)
--- PASS: TestAccAWSCognitoUserPool_withPasswordPolicy (80.81s)
--- PASS: TestAccAWSCognitoUserPool_withLambdaConfig (134.46s)
--- PASS: TestAccAWSCognitoUserPool_withSchemaAttributes (93.93s)
--- PASS: TestAccAWSCognitoUserPool_withVerificationMessageTemplate (75.83s)
--- PASS: TestAccAWSCognitoUserPool_update (150.33s)